base_rest/models/rest_service_registration.py

```diff
@@ -160,10 +163,29 @@ def fix(self):
         for name, method in _inspect_methods(self._service.__class__):
             if not self._is_public_api_method(name):
                 continue
-            if not hasattr(method, "routing"):
+            if not hasattr(method, ROUTING_DEC):
                 methods_to_fix.append(method)
         for method in methods_to_fix:
             self._fix_method_decorator(method)
+        self._update_auth_method_controller()
+
+    def _update_auth_method_controller(self):
+        """
+        During definition of new controller, the _default_auth should be
+        applied on every routes (cfr @route odoo's decorator).
+        This auth attribute should be applied only if the route doesn't already
+        define it.
+        :return:
+        """
+        # If the controller class doesn't have the _default_auth, we don't
+        # have to define it on every routes.
+        if not hasattr(self._controller_class, "_default_auth"):
+            return
+        for name, method in _inspect_methods(self._controller_class):
```

Comment:
@acsonefho I'm not sure if we should apply the default auth méthod. If someone add a route by hand, IMO it's its responsibility to specify the right auth on the route. What do you think about?

Reply:
OTOH it makes sense that if you specify it at controller level, it's applied too all endpoints.
And below, it checks if `auth` is specified, if yes, it's kept. No?

Reply:
@simahawk Yes. If the `auth` is specified on the `@route(...)`, it's kept.

Reply:
I split this PR. So the fix on the route is now here: https://github.com/OCA/rest-framework/pull/130